python-spec/src/somacore/data.py

```diff
@@ -65,7 +76,48 @@ def read(
     ) -> "ReadIter[pa.Table]":
         """Reads a user-defined slice of data into Arrow tables.
 
-        TODO: Further per-param documentation.
+        :param coords: for each index dimension, which rows to read.
+            Defaults to ``None``, meaning no constraint -- all IDs.
+        :param column_names: the named columns to read and return.
+            Defaults to ``None``, meaning no constraint -- all column names.
+        :param partitions: an optional ``ReadPartitions`` hint to indicate
+            how results should be organized.
+        :param result_order: order of read results.
+            This can be one of 'row-major', 'col-major', or 'auto'.
+        :param value_filter: an optional [value filter] to apply to the results.
+            Defaults to no filter.
+
+        **Indexing:**
+
+        Indexing is performed on a per-column basis for each indexed column.
+        To specify dimensions:
+
+        - A sequence of coordinates is accepted, one per indexed dimension.
+        - The sequence length must be less than or equal to the number of
+          indexed dimensions.
+        - If the sequence is shorter than the number of indexed coordinates,
+          then no constraint (i.e. ``None``) is used for the remaining
+          indexed dimensions.
+        - Specifying ``None`` or an empty sequence (e.g. ``()``) represents
+          no constraints over any dimension, returning the entire dataset.
+          TODO: https://github.com/single-cell-data/TileDB-SOMA/pull/910
+
+        Each dimension may be indexed as follows:
+
+        - ``None`` or ``slice(None)`` places no constraint on the dimension.
+        - Coordinates can be specified as a scalar value, a Python sequence
+          (``list``, ``tuple``, etc.), a ``ndarray``, an Arrow array, and
```

Comment:
a `ndarray` -> an `ndarray` (the word is prounced _enn dee uh ray_ which begins with a vowel sound, so euphony requires _an_)